include/qore/intern/qore_socket_private.h

```diff
@@ -468,28 +469,29 @@ struct qore_socket_private {
    }
 
    // returns a new socket
-   DLLLOCAL int accept_internal(SocketSource *source, int timeout_ms = -1, ExceptionSink* xsink = 0) {
+   DLLLOCAL int accept_internal(ExceptionSink* xsink, SocketSource *source, int timeout_ms = -1) {
+      assert(xsink);
       if (sock == QORE_INVALID_SOCKET) {
-	 if (xsink)
-	    xsink->raiseException("SOCKET-NOT-OPEN", "socket must be opened, bound, and in a listening state before new connections can be accepted");
+         // FIXME: remove check
+         if (xsink)
+            xsink->raiseException("SOCKET-NOT-OPEN", "socket must be opened, bound, and in a listening state before new connections can be accepted");
 	 return QSE_NOT_OPEN;
       }
       if (in_op >= 0) {
          if (in_op == gettid()) {
-            if (xsink)
-               se_in_op("Socket", "accept", xsink);
+            se_in_op("Socket", "accept", xsink);
             return QSE_IN_OP;
```

Comment:
the check was adde to `se_in_op()` and all error functions instead to make it easier to remove later -
```
void se_in_op(const char* cname, const char* meth, ExceptionSink* xsink) {
   assert(xsink);
   // FIXME: remove check
   if (xsink)
      xsink->raiseException("SOCKET-IN-CALLBACK", "calls to %s::%s() cannot be made from a callback on an operation on the same socket", cname, meth);
}
```